src/Tribe/Aggregator/Migrate.php

```diff
@@ -0,0 +1,434 @@
+		$response->statuses = $status;
+
+		wp_send_json( $response );
+	}
+
+	/**
+	 * Method that Handles the AJAX converting of Legacy iCal Settings
+	 * AJAX methods will not return anything, only print a JSON string
+	 *
+	 * @return void
+	 */
+	public function ajax_convert_ical_settings() {
+		$response = (object) array(
+			'status' => false,
+			'text' => esc_html__( 'Error, a unknown bug happened and it was impossible to migrate the old iCal Settings to Event Aggregator, try again later.', 'the-events-calendar' ),
+		);
+
+		$post_type = get_post_type_object( Tribe__Events__Main::POSTTYPE );
+
+		if ( empty( $post_type->cap->edit_posts ) || ! current_user_can( $post_type->cap->edit_posts ) ) {
+			$response->status = false;
+			$response->text = esc_html__( 'You do not have permission to migrate the iCal Facebook Settings to Event Aggregator', 'the-events-calendar' );
+
+			wp_send_json( $response );
+		}
+
+		$settings = $this->get_ical_setting();
+
+		$status = (object) array(
+			'error' => array(),
+			'success' => array(),
+		);
+
+		$origin = 'ical';
+
+		foreach ( $settings->imports as $time => $import ) {
+			$import = (object) $import;
+			$record = Tribe__Events__Aggregator__Records::instance()->get_by_origin( $origin );
+
+			/**
+			 * @todo Include the Deactivated logic
+			 */
+			$meta = array(
+				'origin'       => $origin,
+				'type'         => 'schedule',
+				'frequency'    => $this->convert_ical_frequency( $import->schedule ),
+				'file'         => null,
+				'keywords'     => $import->keywords,
+				'location'     => $import->location,
+				'start'        => $import->start,
+				'radius'       => $import->radius,
+				'source'       => $import->url,
+				'content_type' => null,
+				'is_legacy'    => true,
+				'import_id'    => null,
+				'post_status'  => $import->post_status,
+				'category'     => $import->import_category,
+			);
+
+			$post = $record->create( 'schedule', array(), $meta );
+
+			if ( is_wp_error( $post ) ) {
+				$status->error[] = $post;
+			} else {
+				$status->success[] = $post->id;
+
+				// Update status from Draft to Schedule
+				$args['ID'] = absint( $post->id );
+				$args['post_status'] = Tribe__Events__Aggregator__Records::$status->schedule;
+				wp_update_post( $args );
+			}
+		}
+
+		/**
+		 * @todo Create a real Logic for Messaging what happened
+		 */
+		$response->status = true;
+		$response->text = esc_html__( 'Succesfully imported the iCal Settings into Aggregator Records.', 'the-events-calendar' );
+		$response->statuses = $status;
+
+		wp_send_json( $response );
+	}
+
+	/**
+	 * Get the iCal frequency and convert to EA
+	 *
+	 * @param  string $frequency iCal Frequency
+	 * @return string            EA Frequency
+	 */
+	private function convert_ical_frequency( $frequency ) {
+		$results = Tribe__Events__Aggregator__Cron::instance()->get_frequency( array( 'id' => $frequency ) );
+
+		// Return to the Default
+		if ( empty( $results ) ) {
+			return 'every30mins';
```

Comment:
this is the iCal Importer default? The new EA default is daily, should we slide those that way, or is this really the appropriate option?